qcodes/dataset/descriptions/dependencies.py

```diff
@@ -257,6 +269,20 @@ def _to_dict(self) -> InterDependencies_Dict:
                                           "standalones": standalones}
         return output
 
+    def _empty_data_dict(self) -> Dict[str, Dict[str, np.ndarray]]:
+        output: Dict[str, Dict[str, np.ndarray]] = {}
+        for key, value in self.dependencies.items():
+            ps_id = self._paramspec_to_id[key]
+            output[ps_id] = {ps_id: np.array([])}
+            dep_ids = [self._paramspec_to_id[ps] for ps in value]
+            for dep_id in dep_ids:
+                output[ps_id][dep_id] = np.array([])
+        standalones = [self._paramspec_to_id[ps] for ps in self.standalones]
+        for standalone in standalones:
+            output[standalone] = {}
+            output[standalone][standalone] = np.array([])
```

Comment:
Should add a test that the cache keys are the same before and after inserting data into the cache. Especially the standalone keys should be covered. Also check for different datatypes (e.g. the empty arrays here have a default type which should not matter)

Reply:
Added tests of cache of standalones and an assert to trigger if non expected parameters 
are added to the cache